src/index.ts

```diff
@@ -126,9 +126,12 @@ function main(state: State, secrets: Secrets) {
 		log(String(reason), "error");
 	});
 
-	// Cleanup resources on exit.
-	exitHook(() => {
-		client.guilds.cache.get(config.guildId)?.commands.set([]).catch(console.error.bind(console));
+	exitHook(callback => {
+		client.guilds.cache
+			.get(config.guildId)
+			?.commands.set([])
+			.then(callback)
+			.catch(err => log(err, "error"));
```

Comment:
async-await-hook seems to expect you to call the callback in all exit paths if it's specified, so it should also be called in the function passed to `catch`
```suggestion
			.catch(err => { log(err, "error"); callback(); });
```